install/kubernetes/cilium/charts/config/templates/configmap.yaml

```diff
@@ -77,7 +77,11 @@ data:
   #   backend. Upgrades from these older cilium versions should continue using
   #   the kvstore by commenting out the identity-allocation-mode below, or
   #   setting it to "kvstore".
+{{- if .Values.global.etcd.enabled }}
+  identity-allocation-mode: kvstore
+{{- else }}
   identity-allocation-mode: {{ .Values.global.identityAllocationMode }}
+{{- end }}
```

Comment:
Unless we shift around how this is configured, I'm not sure there's a better way. Currently it's by default set to `crd` via the global config map, so we can't detect that it's *not* configured then apply only in that case.

I'm fine with forcing kvstore mode in this way even though it overrides the user-specified configuration, because I don't see a reason why users would want to use the kvstore but manage identities via CRDs. :+1:

Reply:
>because I don't see a reason why users would want to use the kvstore but manage identities via CRDs.

@joestringer `cilium-etcd-operator` won't have a security identity because of the chicken-egg problem.

Reply:
> > because I don't see a reason why users would want to use the kvstore but manage identities via CRDs.
> 
> @joestringer `cilium-etcd-operator` won't have a security identity because of the chicken-egg problem.
So @joestringer what do you think, can we leave it this way or do we want to change this.

Reply:
@aanm isn't that what fixed identities are for?

Reply:
@joestringer it's not 100% reliable, I mean it would be less reliable than CRD based identity.